lambda/es-cleanup.py

```diff
@@ -0,0 +1,210 @@
+            credential_resolver = create_credential_resolver(get_session())
+            credentials = credential_resolver.load_credentials()
+            SigV4Auth(credentials, 'es', es_region).add_auth(req)
+
+            try:
+                preq = req.prepare()
+                session = Session()
+                res = session.send(preq)
+                if res.status_code >= 200 and res.status_code <= 299:
+                    # print("%s %s" % (res.status_code, res.content))
+                    return json.loads(res.content)
+                else:
+                    raise ES_Exception(res.status_code, res._content)
+
+            except ES_Exception as e:
+                if (e.status_code >= 500) and (e.status_code <= 599):
+                    retries += 1  # Candidate for retry
+                else:
+                    raise  # Stop retrying, re-raise exception
+
+    def send_error(self, msg):
+        """Send SNS error
+
+        Args:
+            msg (str): error string
+
+        Returns:
+            None
+        """
+        _msg = "[%s][%s] %s" % (self.name, self.cur_account, msg)
+        print(_msg)
+        if self.cfg["sns_alert"] != "":
+            sns_region = self.cfg["sns_alert"].split(":")[4]
+            sns = boto3.client("sns", region_name=sns_region)
+            sns.publish(TopicArn=self.cfg["sns_alert"], Message=_msg)
+
+    def delete_index(self, index_name):
+        """ES DELETE specific index
+
+        Args:
+            index_name (str): Index name
+
+        Returns:
+            dict: ES answer
+        """
+        return self.send_to_es(index_name, "DELETE")
+
+    def get_indices(self):
+        """ES Get indices
+
+        Returns:
+            dict: ES answer
+        """
+        return self.send_to_es("/_cat/indices")
+
+
+def lambda_handler(event, context):
+    """Main Lambda function
+    Args:
+        event (dict): AWS Cloudwatch Scheduled Event
+        context (object): AWS running context
+    Returns:
+        None
+    """
+    es = ES_Cleanup(event, context)
+    # Index cutoff definition, remove older than this date
+    earliest_to_keep = datetime.date.today() - datetime.timedelta(
+        days=int(es.cfg["delete_after"]))
+    for index in es.get_indices():
+        if index["index"] == ".kibana":
+            # ignore .kibana index
+            print("Found .kibana index - ignoring")
+            continue
+
+        idx_name = '-'.join(word for word in index["index"].split("-")[:-1])
+        idx_date = index["index"].split("-")[-1]
+        print("Found index: %s - %s" % (idx_name, idx_date))
+        if idx_name in es.cfg["index"] or "all" in es.cfg["index"]:
+
+            if idx_date <= earliest_to_keep.strftime(es.cfg["index_format"]):
+                print("Deleting index: %s" % index["index"])
+                es.delete_index(index["index"])
+
+
+if __name__ == '__main__':
+    event = {
+        'account': '123456789012',
```

Comment:
what are these hardcoded values for `account`, `id` and `arn`?

Reply:
this is in func main with an example event just for running locally and doesn't get run by Lambda.